docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
+* `execute`: Method to increase quantity of the specified product.
+  * `execute` calls `decreaseAmount` in the ProductList class.
+  * `decreaseAmount` in ProductList class will call `updateDecreaseQuantity` in the Quantity class.
+
+The following sequence diagram shows how the OutflowCommand works.
+![OutflowCommand Class](images/OutflowCommandSequence.png)
+
+### Delete product feature
+
+#### Implementation
+The delete function is mainly facilitated by `DeleteCommand`. It extends from `Command`.
+
+**Attributes**
+`pid` The unique Product ID for the product to be deleted from the StockPal inventory `productList`
+
+**Methods**
+* `DeleteCommand`: Constructor for creating a new instance of the DeleteCommand class.
+* `execute`: Method to delete `product` with PID `pid` from StockPal's `productList`.
+* `ProductList#deleteProduct`: Method called by `execute` to delete product in `productList`.
+
+Given below is an example usage scenario and how the delete function behaves at each step. The scenario assumes that the
+user has a `product` with `pid` of 2 in `StockPal`'s `productList`.
+
+Step 1. The user executes `delete 2` command to delete a specific `product` with `pid` of 2.
+Step 2. The product is successfully deleted from the inventory list.
+
+The following sequence diagram summarizes what happens when a user inputs a valid `delete` command.
+
+<img src="images/DeleteCommandSequence.png" alt=""/>
+
+### Find product feature
+
+The FindCommand class is responsible for finding a new product to the inventory in the StockPal application.
+
+#### Class diagram of FindCommand:
+<img src="images/FindCommandClass.png" alt=""/>
+
+
+**Attributes**
+* keyword: The name of the product to look for.
+
+**Methods**
+* `FindCommand`: Constructor for creating a new instance of the FindCommand class.
+* `execute`: Method to find the keyword in the productList
+
+
+The following sequence diagram shows how a find operation works when the user inputs a valid `find` command:
+<img src="images/FindCommandSequence.png" alt="FindCommandSequence.png"/>
+
+The following sequence diagram details how `FindCommand#execute()` functions.
+<img src="images/FindCommandExecuteSequenceDiagram.png" alt="FindCommandExecuteSequenceDiagram.png"/>
+
+
+### History product feature
+
+The HistoryCommand class is responsible for finding any inflows or outflows for a particular
+PID in the StockPal application.
+
+#### Class diagram of HistoryCommand:
+<img src="images/HistoryCommandClass.png" alt=""/>
+
+
+**Attributes**
+* pid: The pid of the product to look for.
+
+**Methods**
+* `HistoryCommand`: Constructor for creating a new instance of the HistoryCommand class.
+* `execute`: Method to find the list of transactions for the particular pid
+
+
+The following sequence diagram shows how a find operation works when the user inputs a valid `history` command:
+<img src="images/HistoryCommandSequence.png" alt="HistoryCommandSequence.png"/>
```

Comment:
good color coding for this

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/fa3bac92-a96c-4ee1-8cb4-9585013d0864)